### PR TITLE
Convert method config to a struct for use in the fast path.

### PR DESCRIPTION
**Pull request by markdroth**
This addresses the last comment from @ctiller in #8303.  It is part of addressing #7818.

### CODE REVIEW

File: src/core/ext/client_config/method_config.h [hunk 1]
```diff
@@ -106,11 +106,23 @@ int grpc_method_config_table_cmp(const grpc_method_config_table* table1,
...
+/// grpc_method_config_table_convert().
+void* grpc_method_config_table_get(const grpc_mdstr_hash_table* table,
+                                   const grpc_mdstr* path);
 
 /// Returns a channel arg containing \a table.
 grpc_arg grpc_method_config_table_create_channel_arg(
     grpc_method_config_table* table);
 
+/// Generates a new table from \a table whose values are converted to a
+/// new form via the \a convert_value function.  The new table will use
```
Comment (dgquintas) [inline src/core/ext/client_config/method_config.h]: what new form? Can you add some more detail here about what `convert_value` is expected to do?

Comment (markdroth) [inline src/core/ext/client_config/method_config.h]: Done.

File: src/core/ext/client_channel/client_channel.c [hunk 2]
```diff
@@ -60,6 +60,50 @@
 
 /* Client channel implementation */
 
+typedef enum {
```
Comment (ctiller) [inline src/core/ext/client_channel/client_channel.c]: Add a big ol' block comment:

```
 /*************************************************************************
   * METHOD-CONFIG TABLE
   */

Comment (markdroth) [inline src/core/ext/client_channel/client_channel.c]: Done.

**Comment by ctiller** [comment]
Merge at will.

**Pull request merged**